jujugui/static/gui/src/app/app.js

```diff
@@ -762,55 +771,27 @@ YUI.add('juju-gui', function(Y) {
         }
         this._renderLoginOutLink();
         console.log('successfully logged into controller');
-        // After logging in trigger the app to dispatch to re-render the
-        // components that require an active connection to the controllerAPI.
-        this.dispatch();
         // If the user is connected to a model then the modelList will be
         // fetched by the modelswitcher component.
         if (this.env.get('modelUUID')) {
           return;
         }
         const modelUUID = this._getModelUUID();
-        // If the model UUID provided by the external system is "disconnected"
-        // that means that the external GUI host decided that it wants to load
-        // the GUI in the disconnected mode. In this case, it's not necessary
-        // to switch the model, but we still want to render the breadcrumb for
-        // the new logged in user.
-        if (modelUUID === DISCONNECTED_UUID) {
-          this._renderBreadcrumb();
-          return;
+        const current = this.state.current;
+        if (modelUUID) {
+          // A model uuid was defined in the config so attempt to connect to it.
+          this._listAndSwitchModel(null, modelUUID);
+        } else if (entityPromise !== null) {
+          this._disambiguateUserState(entityPromise);
+        } else if (current && !current.root) {
+          // Drop into disconnected mode and show the profile but only if there
+          // is no state defined in root. If there is a state defined in root
+          // then we want to let that dispatcher handle the routing.
+          this.maskVisibility(false);
+          this.state.changeState({
+            profile: this._getAuth().rootUserName
```

Comment:
We could, but it shouldn't be possible as one of the conditionals in `_getAuth()` is to get the credentials from the controller, and we're in the login callback handlers where it's impossible to get if there are no credentials.